### ISSUE TITLE
Specify desired behavior of CurationConcerns::VersioningService

### ISSUE THREAD

**Issue by flyingzumwalt** [pulibrary/curation_concerns#24]
@mjgiarlo @awead @cam156 et al Does anyone have opinions about `Sufia::VersioningService` (which would become `CurationConcerns::VersioningService`)?  The way I wrote `Hydra::Works::AddFileToGenericFile` it automatically adds a version to the VersionGraph if original_file is marked versionable.  This means that if `CurationConcerns::GenericFileActor` also calls `VersioningService.create` you will get two versions every time you update a GenericFile.  Should I 
 1. ditch the VersioningService code entirely and rely on the AddFile Service from hydra-works* 
 2. change the way `Works::AddFileToGenericFile` works, allowing you to tell it _not_ to create a version even though the file is versionable, so that we can add it later with the VersioningService
 3. implement a `CurationConcerns::AddFileToGenericFile` service that uses the VersionCommitter and use that instead of `Hydra::Works::AddFileToGenericFile`
 4. re-work `CurationConcerns::VersioningService` to _only_ record the VersionCommitter record in the database, and assume that the actual version has been added to Fedora elsewhere by the AddFile service

*The main extra thing that Sufia's VersioningService does is that, in addition to creating a version in Fedora, it records the commit in an ActiveRecord table as a VersionCommitter object.

**Comment by mjgiarlo** [comment]
Sufia's `VersionCommitter` functionality was needed atop Fedora 3 to keep a record of what user created a version. Is this still needed in Fedora 4, or does it keep that information itself? I suspect we no longer need `VersionCommitter`.

**Comment by grosscol** [comment]
My understanding is that this is a question of to have the framework create the version or to have the application do it.  

1. Relying entirely on the AddFile Service from Hydra works is workable, but then requires some logic to receive that a version was created and stored on the fedora side and make the corresponding update to the mysql table.
2.  Is there a situation where one would not want to add a version to the VersionGraph even when the original_file is marked versionable?  If that is the case, and it's not going to be a frequent occurrence, option 2 seems the least intrusive.
3.  I would rather use more common framework that less.
4. A version committer that only records changes to the sql backend would have to have logic to prevent it from making records of versions that were not made in fedora.

In short, I'd like to go with option 2.

**Comment by mjgiarlo** [comment]
I'm leaning towards option 2, myself, though I don't have a use case for *not* creating a version, so I'd also be :ok: with option 1. I'd be interested in hearing what others think. @jpstroop @terrellt @grosscol @hectorcorrea @jcoyne @escowles @pbinkley @rotated8 @elrayle

**Comment by flyingzumwalt** [comment]
To follow up and clarify one of @mjgiarlo's comments - the only reason why we create a VersionCommitter object in the sql database is because Fedora 3 did not let us record info about _who_ made each version.  If it's possible to add that info into Fedora, then Option 5 would be to make Works::AddFile service accept an option telling it who to record as the committer when it creates the version. (and then update ActiveFedora::VersionResource to return that info when requested)

**Comment by mjgiarlo** [comment]
:+1:

**Comment by jpstroop** [comment]
I think option 2 or 5. I haven't been tracking `HW` that closely this week, but I _think_ that if 5 is feasible then that's the way to go.

**Comment by flyingzumwalt** [comment]
Looks like [ActiveFedora::VersionsGraph::ResourceVersion](https://github.com/projecthydra/active_fedora/blob/master/lib/active_fedora/versions_graph.rb#L34-L36) does not have an attribute for tracking the committer.  It might be possible to change that, and it's probably a good idea, but I can't muck around in ActiveFedora any more this week.  That rules out Option 5 for this week.

... so sounds like it's option 2 for now?  Any opinions?

**Comment by mjgiarlo** [comment]
@flyingzumwalt We might could ask the conclave if they have any cycles to tackle this.

**Comment by flyingzumwalt** [comment]
For now, I'm going with Option 2 in curation_concerns.  If anyone (@mjgiarlo?, @jpstroop?) wants option 5 to happen in the future, please submit a ticket on AF requesting that ResourceVersion should track a :committer attribute and Versionable#create_version should accept an optional `committer:` keyword argument.
